_posts/2024-12-17-release-0_9_0.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title:  "kroxylicious release 0.9.0"
+date:   2024-12-17 00:00:00 +0000
+author: "Sam Barker"
+author_url: "https://github.com/SamBarker"
+categories:  [releases, kroxylicious, kafka3.9]
+---
+
+The Kroxylicious project is very pleased to announce the [0.9.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.9.0) release of Kroxylicious. This release introduces support for Apache Kafka 3.9. 
+
+We generally expect Kroxylicious to be both forwards and backwards compatible across Kafka version just like the Apache Kafka client and broker. While we tested with pre-release versions Kafka 3.9 we discovered after it was released that we had missed a case in the protocol version negotiation. When the API Versions request version was updated we were unable to decode the response. In the 0.9.0 release we adopt the same behaviour as the Kafka broker and respond with API version 0 if we do not support the protocol version requested by the client - we have also expanded our test coverage to get earlier warnings of similar issues in the future. 
+
+This release also marks the introduction of our new connection handling sate machine at the core of the proxy. We have evolved state machine at the core of the proxy to separate out the state carried at various stages of the connection life cycle and provide stronger rules around transitions between states. This is still a work in progress as its implications are rather wide-ranging however with think it gives us a stronger core to base everything else around. This does for the moment limit our ability to offer authentication offload support, we do intend to restore this eventually (please do get in touch if this use case is important to you).
```

Comment:
```suggestion
This release also marks the introduction of our new connection handling state machine at the core of the proxy. We have evolved the state machine at the core of the proxy to separate out the state carried at various stages of the connection life cycle and provide stronger rules around transitions between states. This is still a work in progress as its implications are rather wide-ranging however we think it gives us a stronger core to base everything else around. This does, for the moment, limit our ability to offer authentication offload support, we do intend to restore this eventually (please do get in touch if this use case is important to you).
```

Reply:
To add to this:
#### Spelling and grammar:

```suggestion
This release also marks the introduction of our new connection handling state machine at the core of the proxy. We have evolved the state machine at the core of the proxy to separate out the state carried at various stages of the connection lifecycle, and to provide stronger rules around transitions between states. This is still a work-in-progress as its implications are rather wide-ranging, however we think it gives us a stronger core to base everything else around. This does, for the moment, limit our ability to offer authentication offload support. We do intend to restore this eventually, so please get in touch if this use case is important to you.
```

#### Content:

> This release also marks the introduction of our new connection handling state machine at the core of the proxy.

It was hard for me to tell at first read whether `connection handling state machine` is meant to mean "a state machine that is handling connections" (i.e. `state machine` is a noun and `connection handling` is a verb), or if it's the full name of the thing, (i.e. `Connection Handling State Machine` is a proper noun, so each word should be capitalised). I think it would be less confusing overall to call it the `connection handler state machine` here, which makes it a bit more obvious what we're talking about.

> We have evolved the state machine at the core of the proxy to separate out the state carried at various stages of the connection lifecycle, and to provide stronger rules around transitions between states.

Using the phrasing `state machine at the core of the proxy` here right after using it in the previous sentence feels a bit repetitive. I actually think it fits better here than in the previous sentence, so I would just remove the `at the core of the proxy` part from the previous sentence and leave this one be.

> This is still a work-in-progress as its implications are rather wide-ranging, however we think it gives us a stronger core to base everything else around.

Rephrase to: "This is still a work-in-progress due to the wide-ranging implications it has, however we think it gives us a stronger core to base everything else around."

> This does, for the moment, limit our ability to offer authentication offload support. We do intend to restore this eventually, so please get in touch if this use case is important to you.

Would be good to describe or link to where users can get in touch with us here.